### ISSUE TITLE
[a11y]: focus not visible when clicking on combobox toggle

### ISSUE THREAD

**Issue by grahamharper** [carbon-design-system/carbon#9776]
### Package

carbon-components-react

### Browser

Chrome

### Operating System

MacOS

### Package version

v10.44.2

### Automated testing tool and ruleset

none

## Current
When clicking with a mouse on the combobox toggle, there is no visual indication of focus.
![Kapture 2021-09-30 at 12 13 18](https://user-images.githubusercontent.com/2853295/135445178-684ba021-8982-4336-95a5-5ac086c2120f.gif)

## Expected
The input should receive a visual focus when clicking the toggle. This was the behaviour in a prior version of Carbon (10.17 and likely later versions)

![Kapture 2021-09-30 at 12 15 35](https://user-images.githubusercontent.com/2853295/135445399-d051ef50-1355-4590-946a-9c5dec6aa56e.gif)

### WCAG 2.1 Violation

Success Criterion 2.4.7: Focus Visible

### CodeSandbox example

https://codesandbox.io/s/cranky-mestorf-nevdm

### Steps to reproduce

Click on the chevron/toggle with a mouse to expand the listbox.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by Spectre-ak** [comment]
Can I work on this?
